Donejs Init Fails: Cannot find module 'yeoman-environment' (1)

When running donejs init, initialization fails, with the error:

```
└── yeoman-generator@0.20.2 (read-chunk@1.0.1, detect-conflict@1.0.0, yeoman-welcome@1.0.1, path-is-absolute@1.0.0, path-exists@1.0.0, xdg-basedir@1.0.1, yeoman-assert@2.0.0, user-home@1.1.1, rimraf@2.4.2, diff@1.4.0, text-table@0.2.0, mime@1.3.4, dargs@4.0.1, istextorbinary@1.0.2, async@0.9.2, debug@2.2.0, chalk@1.1.1, nopt@3.0.3, run-async@0.1.0, mkdirp@0.5.1, shelljs@0.5.3, through2@0.6.5, cli-table@0.3.1, dateformat@1.0.11, pretty-bytes@1.0.4, glob@5.0.14, underscore.string@3.1.1, findup-sync@0.2.1, mem-fs-editor@2.0.4, cross-spawn@1.0.4, github-username@2.0.0, class-extend@0.1.1, download@4.2.0, gruntfile-editor@1.0.0, sinon@1.16.0, html-wiring@1.2.0, inquirer@0.8.5)
Error: Cannot find module 'yeoman-environment'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at /usr/local/lib/node_modules/donejs/bin/donejs:56:20
    at _fulfilled (/usr/local/lib/node_modules/donejs/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/usr/local/lib/node_modules/donejs/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/donejs/node_modules/q/q.js:796:13)
    at /usr/local/lib/node_modules/donejs/node_modules/q/q.js:604:44
    at runSingle (/usr/local/lib/node_modules/donejs/node_modules/q/q.js:137:13)
```

NOTE: This issue was resolved by installing yeoman globally - `npm i yo -g`

--- a/.jshintrc
+++ b/.jshintrc
@@ -1,0 +1,27 @@
+{
+  "node": true,
+  "esnext": true,
+  "bitwise": true,
+  "camelcase": true,
+  "curly": true,
+  "eqeqeq": true,
+  "immed": true,
+  "indent": 2,
+  "latedef": "nofunc",
+  "newcap": false,
+  "noarg": true,
+  "regexp": true,
+  "undef": true,
+  "unused": true,
+  "strict": false,
+  "trailing": true,
+  "smarttabs": true,
+  "white": false,
+  "globals": {
+    "it": true,
+    "describe": true,
+    "before": true,
+    "after": true,
+    "exports": true
+  }
+}

--- a/bin/donejs
+++ b/bin/donejs
@@ -1,7 +1,6 @@
 #!/usr/bin/env node
 
 var path = require('path');
-var appModule = require('app-module-path');
 var fs = require('fs');
 var program = require('commander');
 var mkdirp = require('mkdirp');
@@ -51,17 +50,15 @@ var actions = {
     }
 
     root = path.join(process.cwd(), 'node_modules');
-    appModule.addPath(root);
 
-    log(utils.generate([ 'app', {
+    log(utils.generate(root, [ 'app', {
         version: mypkg.version,
         packages: mypkg.donejs
       }
     ]));
   },
   generate: function(type, options) {
-    appModule.addPath(root);
-    log(utils.generate([ [type].concat(options) ]));
+    log(utils.generate(root, [ [type].concat(options) ]));
   },
   catchAll: function(command) {
     console.error('Could not run `' + command + '`');

--- a/lib/utils.js
+++ b/lib/utils.js
@@ -1,6 +1,8 @@
 var Q = require('q');
+var path = require('path');
 var npm = require('npm');
 var exec = require('child_process').exec;
+var yeoman = require('yeoman-environment');
 
 // Returns a .then-able function that installs a single module
 // if it is not available in the current path
@@ -16,7 +18,7 @@ var installIfMissing = exports.installIfMissing = function(module) {
     }
 
     return previous;
-  }
+  };
 };
 
 // Run a command and pipe the output.
@@ -37,8 +39,8 @@ var runCommand = exports.runCommand = function(cmd, args) {
 
   child.on('exit', function(status) {
     if(status !== 0) {
-      deferred.reject(new Error('Command `' + cmd + ' '
-        + (args || []).join(' ') + '` did not complete successfully'));
+      deferred.reject(new Error('Command `' + cmd + ' ' +
+        (args || []).join(' ') + '` did not complete successfully'));
     } else {
       deferred.resolve(child);
     }
@@ -48,17 +50,16 @@ var runCommand = exports.runCommand = function(cmd, args) {
 };
 
 // Run any of the Yeoman generators from the current generator-donejs
-exports.generate = function(args) {
+exports.generate = function(root, args) {
   return Q.ninvoke(npm, 'load', { loaded: false })
-    .then(installIfMissing('yeoman-environment'))
     .then(installIfMissing('generator-donejs'))
     .then(function () {
-      var yeoman = require('yeoman-environment');
       var env = yeoman.createEnv();
-      var generators = require('generator-donejs');
+      var generators = require(path.join(root, 'generator-donejs'));
 
       Object.keys(generators).forEach(function(name) {
-        env.register(require.resolve('generator-donejs/' + name), name);
+          var fullName = path.join(root, 'generator-donejs', name);
+          env.register(require.resolve(fullName), name);
       });
 
       return Q.npost(env, 'run', args);

--- a/package.json
+++ b/package.json
@@ -7,7 +7,9 @@
     "donejs": "./bin/donejs"
   },
   "scripts": {
-    "test": "mocha test/ --timeout 120000",
+    "test": "npm run jshint && npm run mocha",
+    "jshint": "jshint lib/. bin/donejs test/. --config",
+    "mocha": "mocha test/ --timeout 120000",
     "verify": "echo \"NPM script ran\"",
     "document:watch": "documentjs --watch",
     "document": "documentjs -f && rm -rf site/img && cp -R docs/img site/img",
@@ -33,14 +35,15 @@
   "homepage": "https://github.com/donejs/donejs",
   "devDependencies": {
     "documentjs": "^0.3.0-pre.4",
+    "jshint": "^2.8.0",
     "mocha": "^2.2.5"
   },
   "dependencies": {
-    "app-module-path": "^1.0.2",
     "commander": "^2.8.1",
     "mkdirp": "^0.5.1",
     "npm": "^2.13.1",
-    "q": "^1.4.1"
+    "q": "^1.4.1",
+    "yeoman-environment": "^1.2.7"
   },
   "directories": {
     "doc": "docs"

--- a/test/utils.js
+++ b/test/utils.js
@@ -33,7 +33,8 @@ describe('DoneJS CLI tests', function() {
 
     it('generate .component', function(done) {
       var moduleName = 'dummy/component.component';
-      utils.generate([
+      var root = path.join(__dirname, '..', 'node_modules');
+      utils.generate(root, [
           ['component', moduleName, 'dummy-component']
         ])
         .then(function() {

